### PR TITLE
Added information and warning properties to TestContext  #2908

### LINKED ISSUE #2908: TestContext should have test-output properties for Information and Warning
TestContext currently have two properties for test-output, Error and Progress.  
1. Both write to the StdError, but only Error is a real error, Progress is informational and should be written to StdOut
2. There should be two more properties Information and Warning, allowing the user to easily add context information in the test that appears in the test output. 

Note:  I am fully aware that parallel execution makes a mess out of such messages, but this should be up to the user.  If the user have a need for these messages they should be able to use them, for diagnostic purposes or for any reason they like.  

I am adding a corresponding issue to the NUnit3TestAdapter to respond correctly to these 4 streams.  For the adapter it doesn't matter which stream is being used (StdError or StdOut) as it only see the named "streams" and the value being sent.

### PR DESCRIPTION
**Pull request by OsirisTerje**
Closes #2908

### CODE REVIEW

File: src/NUnitFramework/tests/TextOutputTests.cs [hunk 1]
```diff
@@ -92,6 +92,8 @@ public void MultipleWrites()
             // Test purely for display purposes
             TestContext.Progress.WriteLine("TestContext.Progress displays immediately");
             TestContext.Error.WriteLine("TestContext.Error displays immediately as well");
+            TestContext.Information.WriteLine("TestContext.Information displays immediately as well");
+            TestContext.Warning.WriteLine("TestContext.Warning displays immediately as well");
```
Comment (CharliePoole) [inline src/NUnitFramework/tests/TextOutputTests.cs]: Most runners won't actually display these.

**Comment by CharliePoole** [comment]
I've discussed on the issue why I think this feature is a mistake. Since I no longer formally review changes to the framework, I won't make an exception here by requesting changes. I'll just say that this requires changes to every runner and invites mire issue reports about why it sort of looks like conditional logging but really isn't.

**Comment by CharliePoole** [comment]
No issue reference.

**Comment by rprouse** [comment]
This has been dormant for three years, closing.

**Pull request closed without merging**